modules/dnn/src/layers/elementwise_layers.cpp

```diff
@@ -1400,6 +1400,137 @@ struct PowerFunctor : public BaseFunctor
     int64 getFLOPSPerElement() const { return power == 1 ? 2 : 10; }
 };
 
+struct ExpFunctor : public BaseFunctor
+{
+    typedef ExpLayer Layer;
+    float base, scale, shift;
+
+    ExpFunctor(float base_ = -1.f, float scale_ = 1.f, float shift_ = 0.f)
+        : base(base_), scale(scale_), shift(shift_)
+    {
+        CV_Assert(base == -1.f || base > 0.f);
+    }
+
+    bool supportBackend(int backendId, int targetId)
+    {
+        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_CUDA ||
+               backendId == DNN_BACKEND_HALIDE || backendId == DNN_BACKEND_INFERENCE_ENGINE_NGRAPH;
+    }
+
+    void apply(const float* srcptr, float* dstptr, int len, size_t planeSize, int cn0, int cn1) const
+    {
+        float a = scale, b = shift;
+        if( base == -1.f )
+        {
+            for( int cn = cn0; cn < cn1; cn++, srcptr += planeSize, dstptr += planeSize )
+            {
+                for( int i = 0; i < len; i++ )
+                {
+                    float x = srcptr[i];
+                    dstptr[i] = exp(a*x + b);
+                }
+            }
+        }
+        else
+        {
+            for( int cn = cn0; cn < cn1; cn++, srcptr += planeSize, dstptr += planeSize )
+            {
+                for( int i = 0; i < len; i++ )
+                {
+                    float x = srcptr[i];
+                    dstptr[i] = pow(base, a*x + b);
```

Comment:
We don't really need this code branch.

```
pow(base, exponent) = exp(ln(base) * exponent)  // base > 0
= exp(ln_base * (scale * INPUT + shift))
= exp((ln_base * scale) * INPUT + (ln_base * shift))
= exp(scale_normalized * INPUT + shift_normalized)
```

So add new "normalized" fields :
```
float ln_base = base == -1 ? 1 : ln(base);
float scale_normalized = scale * ln_base;
float shift_normalized = shift * ln_base;
```

and use `exp()` code only.

Reply:
Done. Thanks for pointing it out.